Arrays

arzamaskov opened the pull request:
Добавлено решение для arrays

aarzamaskov committed:
fix: add solution

aarzamaskov committed:
fix: remove "skip" from tests

aarzamaskov committed:
fix: rework replace function without standart functions

aarzamaskov committed:
fix: rework search function using recursion

NickelGit commented on test/exercise/arrays/solution.rb:
```diff
@@ -2,11 +2,25 @@ module Exercise
     class << self
       def replace(array)
-        array
+        max = array.reduce { |larger, item| item > larger ? item : larger }
```
Вынеси пожалуйста функцию поиска максимума в отдельный метод.

aarzamaskov replied:
Fixed

NickelGit commented on test/exercise/arrays/solution.rb:
```diff
@@ -2,11 +2,25 @@ module Exercise
+
+        return middle if query == array[middle]
+
+        if query > array[middle]
+          counter = search(array.slice(middle + 1, right), query)
+          counter == -1 ? -1 : (middle + 1) + counter
+        else
+          search(array.slice(left, middle), query)
+        end
```
Давай вместо slice добавим в аргументы границы поиска, и будем соответственно передавать сам массив, значение которое мы ищем, левую границу поиска и правую границу поиска.
И заменим `if-else` на тернарную операцию.

arzamaskov replied:
Немного непонятно, в функции search - изначально только два параметра, т.е. нужно будет добавить еще два параметра - левую и правую границы?

aarzamaskov replied:
Переделал с использованием лямбда-фукнции, массив и искомое значение передавать в лямбду не стал, т.к. использую замыкания

aarzamaskov committed:
fix: implement code reviewer recomendations:
- max value function export into a separate function
- recursive binary search function rework with lambda
